specs/ContextMenuRequested.md

```diff
@@ -0,0 +1,383 @@
+    /// or add/ remove from the default context menu.
+    [uuid(a1d309ee-c03f-11eb-8529-0242ac130003), object, pointer_default(unique)]
+    interface ICoreWebView2ContextMenuRequestedEventArgs : IUnknown {
+        /// The list of default ContextMenuItem objects
+        [propget] HRESULT MenuItems([out, retval] ICoreWebView2ContextMenuItemCollection ** menuItems);
+
+        /// The coordinates where the right click occured
+        [propget] HRESULT Location([out, retval] POINT* coordinates);
+
+        /// The context that the user selected
+        [propget] HRESULT Context([out, retval] COREWEBVIEW2_SELECTION_CONTEXT* context);
+
+        /// Returns the selected Context Menu Item
+        [propget] HRESULT SelectedItem([out, retval] COREWEBVIEW2_CONTEXT_MENU_ITEM_ID* selectedItem);
+
+        /// Sets the Selected Menu Item to respond to the server
+        [propput] HRESULT SelectedItem([in] COREWEBVIEW2_CONTEXT_MENU_ITEM_ID selectedItem);
+
+        /// Whether the App will draw context menu. False by default, meaning WebView should display default context menu
+        /// If set to true, app developer will handle displaying the Context Menu using the data provided.
+        [propput] HRESULT Handled([out, retval] BOOL* handled);
+
+        /// Sets the Handled property
+        [propget] HRESULT Handled([in] BOOL handled);
+
+        /// Returns an `ICoreWebView2Deferral` object. Use this operation to
+        /// complete the event at a later time.
+        HRESULT GetDeferral([out, retval] ICoreWebView2Deferral ** deferral);
+    }
+```
+
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+
+    runtimeclass CoreWebView2ContextMenuRequestedEventArgs;
+    runtimeclass CoreWebView2ContextMenuItem;
+    runtimeclass CoreWebView2ContextMenuItemCollection;
+    runtimeclass CoreWebView2ContextMenuItemCollectionIterator;
+
+    enum CoreWebView2ContextMenuItemID {
+        /// Custome type, will include Items brought in by browser extensions
+        Custom = 0,
+
+        /// No Command
+        None = 1,
+
+        /// Page Commands
+        Back = 2,
+        Forward = 3,
+        Reload = 4,
+        Stop = 5,
+        NewWindow = 6,
+        Print = 7,
+
+        /// Clipboard Commands
+        Cut = 8,
+        Copy = 9,
+        Paste = 10,
+        PasteAsPlainText = 11,
+        SelectAll = 12,
+
+        /// Saving
+        SaveLinkAs = 13,
+        SaveImageAs = 14,
+        
+        Inspect = 15,
+        ... /// list is not complete
+    };
+
+    enum CoreWebView2SelectionContext
+    {
+        Image = 0,
+        Link = 1,
+        Textbox = 2,
+        Text = 3,
+        Page = 4,
+        Video = 5,
+        PDF = 6,
+        ... /// list is not complete
+    };
+
+    runtimeclass CoreWebView2ContextMenuRequestedEventArgs
+    {
+        Boolean Handled { get; set; }
+        CoreWebView2SelectionContext Context { get; };
+        CoreWebView2ContextMenuItemCollection MenuItems { get; };
```

Comment:
Same thing as the above comment. The reason I did it this way was to allow for the collection to be added to and removed from by the developer in the second piece of this API. If you know of a good workaround for supporting an editable list in .Net/WinRT, that would definitely help my case.

Reply:
In .NET the usual collection classes invalidate outstanding iterators if you modify the underlying collection so I'm not sure that helps. If we only have methods to modify the collection by index its sort of up to the end dev to be aware of how they're modifying the collection as they index over it, but I think that's fine.

It could be an interface with something like:

* Count
* GetValueAtIndex
* RemoveValueAtIndex
* AddValueAtIndex

In .NET we can't use IReadOnlyCollection but probably just ICollection. The implementation for this will be more involved than our existing IReadOnlyCollection conversion though so its mutable. Currently we can just create a new collection, fill it up to match our COM/Win32 collection, and hand off the .NET version. Its still doable of course it will just require more work.